### PR TITLE
Update `ignore` to v5

### PR DESCRIPTION
**Pull request by fisker**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/cli/util.js [hunk 1]
```diff
@@ -443,7 +443,11 @@ function formatFiles(context) {
   }
 
   eachFilename(context, context.filePatterns, filename => {
-    const fileIgnored = ignorer.filter([filename]).length === 0;
+    let fileIgnored = false;
+    try {
+      fileIgnored = ignorer.ignores(path.relative(process.cwd(), filename));
+    } catch (_) {}
```
Comment (fisker) [inline src/cli/util.js]: empty catch not allowed? `eslint-disable` or `change eslintrc` ?

File: src/cli/util.js [hunk 2]
```diff
@@ -443,7 +443,13 @@ function formatFiles(context) {
   }
 
   eachFilename(context, context.filePatterns, filename => {
-    const fileIgnored = ignorer.filter([filename]).length === 0;
+    let fileIgnored = false;
+    try {
+      fileIgnored = ignorer.ignores(path.relative(process.cwd(), filename));
+    } catch (_) {
+      // eslint-disable-line no-empty
```
Comment (lydell) [inline src/cli/util.js]: You can change this to `// Do nothing`. A comment inside a block counts as non-empty in ESLint. This way it’s explicit that you actually intended to ignore the error.

Comment (lydell) [reply src/cli/util.js]: Btw, why _are_ we ignoring errors here?

Comment (fisker) [reply src/cli/util.js]: https://github.com/kaelzhang/node-ignore#upgrade-4x---5x

`ignorer.ignores('../file-from-parent')`

throws

Comment (fisker) [inline src/cli/util.js]:
```suggestion
      // Invalid path
```

File: src/common/get-file-info.js [hunk 3]
```diff
@@ -68,8 +68,15 @@ function _getFileInfo({
   resolveConfig = false,
   sync = false
 }) {
+  let ignored = true;
+  try {
+    ignored = ignorer.ignores(path.relative(process.cwd(), filePath));
+  } catch (_) {
+    // eslint-disable-line no-empty
```
Comment (fisker) [inline src/common/get-file-info.js]:
```suggestion
    // Invalid path
```

File: src/cli/util.js (hunk 2 above)
Comment (fisker) [reply src/cli/util.js]: I did tried `isValidPath`, but they are the same.

Comment (fisker) [reply src/cli/util.js]: > You can change this to `// Do nothing`. A comment inside a block counts as non-empty in ESLint. This way it’s explicit that you actually intended to ignore the error.

thanks

File: src/cli/util.js [hunk 4]
```diff
@@ -443,7 +443,13 @@ function formatFiles(context) {
   }
 
   eachFilename(context, context.filePatterns, filename => {
-    const fileIgnored = ignorer.filter([filename]).length === 0;
+    let fileIgnored = false;
+    try {
+      fileIgnored = ignorer.ignores(path.relative(process.cwd(), filename));
+    } catch (_) {
+      // Invalid path
```
Comment (lydell) [inline src/cli/util.js]: Would it be better to do this update in `next` instead and show this error to the user? I’m not sure what an “invalid path” is, but it sounds useful to get to know that some of my patterns are wrong?

Comment (fisker) [reply src/cli/util.js]: I don't know. I'm not changing current behave

Comment (lydell) [inline src/cli/util.js]:
```suggestion
      // Invalid path. TODO: Should we show this error to the user, rather than swallowing it?
```

Comment (lydell) [reply src/cli/util.js]: @evilebottnawi Do you have experience from ESLint or stylelint?

File: src/cli/util.js [hunk 5]
```diff
@@ -443,7 +443,13 @@ function formatFiles(context) {
   }
 
   eachFilename(context, context.filePatterns, filename => {
-    const fileIgnored = ignorer.filter([filename]).length === 0;
+    let fileIgnored = false;
+    try {
+      fileIgnored = ignorer.ignores(path.relative(process.cwd(), filename));
+    } catch (_) {
+      // Invalid path. TODO: Should we show this error to the user, rather than swallowing it?
```
Comment (evilebottnawi) [inline src/cli/util.js]: Maybe we should add note for next major release about that?

**Comment by fisker** [comment]
Not sure what else I need do

**Comment by fisker** [comment]
After reading  https://github.com/sindresorhus/globby/pull/120 , https://github.com/sindresorhus/globby/pull/112 and  https://github.com/sindresorhus/globby/pull/104 I think  I did something wrong

**Pull request closed without merging**